pandas/core/nanops.py

```diff
@@ -1728,12 +1731,27 @@ def na_accum_func(values: ArrayLike, accum_func, *, skipna: bool) -> ArrayLike:
                 result, dtype=orig_dtype
             )
 
-    elif skipna and not issubclass(values.dtype.type, (np.integer, np.bool_)):
+    elif (
```

Comment:
i think maybe ok to have is_ea_dtype be a separate elif; much easier to rationalize here

Reply:
Could do it like that